### PR TITLE
Remove TemporaryGettingStarted.md.

### PR DESCRIPTION
**Pull request by grynspan**
This PR removes TemporaryGettingStarted.md and references to it, as well as references in documentation to Xcode 16 betas. Now that Swift 6 has shipped, these instructions are no longer needed.

The section about testing on WASI has been moved to Documentation/WASI.md.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,8 +122,8 @@ incrementally, at your own pace.
 Detailed documentation for Swift Testing can be found on the
 [Swift Package Index](https://swiftpackageindex.com/swiftlang/swift-testing/main/documentation/testing).
 There, you can delve into comprehensive guides, tutorials, and API references to
-make the most out of this package. To try it yourself, see
-[Getting Started](https://swiftpackageindex.com/swiftlang/swift-testing/main/documentation/testing/temporarygettingstarted).
+make the most out of this package. Swift Testing is included with the Swift
+toolchain and Xcode 16.
```
Comment (stmontgomery) [inline README.md:126]: Might be worth calling out Swift **6** toolchains

File: Documentation/WASI.md
Comment (stmontgomery) [inline Documentation/WASI.md:1]: We have a sub-`README.md` file for the `Documentation/` directory, can we link to this new file from there?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -41,8 +30,8 @@ and clone the fork which will make opening PRs easier.
 
 ### Using Xcode (easiest)
 
-1. Install the latest beta version of
-   [Xcode](https://developer.apple.com/xcode).
+1. Install Xcode 16 or newer. You can find Xcode on the [Apple Developer](https://developer.apple.com/xcode/)
```
Comment (iamleeg) [inline CONTRIBUTING.md:33]:
```suggestion
1. Install Xcode 16 or newer, from the [Apple Developer](https://developer.apple.com/xcode/)
```

Not a very strong suggestion, I just think it's a bit more straightforward.

**Review by iamleeg — APPROVED** [review]
I haven't got anything to say about WASM because I don't really know it but I think you've handled removing the getting started guide fine. I made one suggestion but it's a nit and I don't mind if you handle it differently or not at all.

File: CONTRIBUTING.md (hunk 2 above)
Comment (grynspan) [reply CONTRIBUTING.md:33]: Sure, will tweak.

**Comment by grynspan** [comment]
@swift-ci test

**Review by stmontgomery — APPROVED** [review]

**Pull request merged**